scripts/pack.sh

```diff
@@ -10,13 +10,16 @@ rm -rf ${DST:?}/* 2> /dev/null || :
 mkdir -p $DST
 
 # Detect version info
-COMMIT=`git rev-parse HEAD`
+COMMIT=`git rev-parse HEAD 2> /dev/null || :`
 VERSION=`cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[\",]//g' | tr -d '[[:space:]]'`
 
 if [ -n "$APPVEYOR_REPO_BRANCH" ]; then
     BRANCH=$APPVEYOR_REPO_BRANCH
-else
+elif [ -n "$COMMIT" ]; then
     BRANCH=`git rev-parse --abbrev-ref HEAD`
+else
+    BRANCH="unknown"
+    COMMIT="unknown"
 fi
```

Comment:
Hmm, this all sounds a bit backwards to me.

Wouldn't it be better if we separated CI-magic in CI-scripts, and instead passed the branch name and commit id into scripts/pack.sh?

I mean, it's much cleaner if we have a set of scripts that are supposed to be called from the CI environment, and a set of scripts that doesn't care where it's called from...

Reply:
Yeah, the branch/commit detection is a bit awkward, but replacing the mechanism will require some more work and we can probably improve things in a future PR.

This PR will as-is fix the immediate problem for users trying to build our source code archives at least...

Reply:
Fair enough.